### PR TITLE
python3Packages.ripe-atlas-sagan: init at 1.3.1

### PR DESCRIPTION
**Pull request by RaitoBezarius**
###### Motivation for this change

I am in the process of packaging https://github.com/RIPE-NCC/ripe-atlas-tools which is a very useful networking tool.
This tool requires this package, which can be useful in other RIPE Atlas-based projects.

I also introduce myself as a maintainer of this package (and the ones which will depend on it, i.e. `socketio-client`, `ripe-atlas-cousteau` and `ripe-atlas-tools` itself in the future PRs).

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/ripe-atlas-sagan/default.nix [hunk 1]
```diff
@@ -0,0 +1,37 @@
+{ lib
+, buildPythonPackage
+, python-dateutil
+, pytz
+, cryptography
+, nose
+, fetchFromGitHub }:
+buildPythonPackage rec {
```
Comment (fabaff) [inline pkgs/development/python-modules/ripe-atlas-sagan/default.nix:8]:
```suggestion
, fetchFromGitHub
}:

buildPythonPackage rec {
```

File: pkgs/development/python-modules/ripe-atlas-sagan/default.nix [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  pname = "ripe-atlas-sagan";
+  version = "1.3.1";
+
+  src = fetchFromGitHub {
+    owner = "RIPE-NCC";
+    repo = "ripe-atlas-sagan";
```
Comment (fabaff) [inline pkgs/development/python-modules/ripe-atlas-sagan/default.nix:14]: Opinionated change:

```suggestion
    repo = pname;
```

File: pkgs/development/python-modules/ripe-atlas-sagan/default.nix [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+  propagatedBuildInputs = [
+    python-dateutil
+    pytz
+    cryptography
+  ];
+
+  checkInputs = [
+    nose
+  ];
```
Comment (fabaff) [inline pkgs/development/python-modules/ripe-atlas-sagan/default.nix:27]: Please add `pythonImportsCheck`. Something like this should check if the module can be imported.

```suggestion
  ];

  pythonImportsCheck = [
    "ripe.atlas.sagan"
  ];
```

File: pkgs/development/python-modules/ripe-atlas-sagan/default.nix [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+  passthru.updateScript = ./update.sh;
```
Comment (fabaff) [inline pkgs/development/python-modules/ripe-atlas-sagan/default.nix:29]: Same as in https://github.com/NixOS/nixpkgs/pull/155269/files#r785733560 applies here.

File: pkgs/development/python-modules/ripe-atlas-sagan/default.nix [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+  meta = with lib; {
+    description = "A parsing library for RIPE Atlas measurements results";
+    homepage = "https://github.com/RIPE-NCC/ripe-atlas-sagan";
+    license = licenses.gpl3;
```
Comment (fabaff) [inline pkgs/development/python-modules/ripe-atlas-sagan/default.nix:34]: There is no mentioning about "+/Plus".

```suggestion
    license = licenses.gpl3Only;
```

**Comment by RaitoBezarius** [comment]
Everything has been addressed, cc @fabaff

**Comment by fabaff** [comment]
Can you squash the commits for the package (but keep the one `maintainer-list.nix`)?

**Comment by RaitoBezarius** [comment]
@fabaff Fixed!

File: pkgs/development/python-modules/ripe-atlas-sagan/default.nix [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+
+  propagatedBuildInputs = [
+    python-dateutil
+    pytz
+    cryptography
+  ];
+
+  checkInputs = [
+    nose
+  ];
```
Comment (jonringer) [inline pkgs/development/python-modules/ripe-atlas-sagan/default.nix:29]: I get:
```
======================================================================
FAIL: tests.test_invalid_country_code
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/nix/store/30bmb2mhbffj86swq6g9kw9vc8aqjgaz-python3.9-nose-1.3.7/lib/python3.9/site-packages/nose/case.py", line 19>
    self.test(*self.arg)
  File "/nix/store/30bmb2mhbffj86swq6g9kw9vc8aqjgaz-python3.9-nose-1.3.7/lib/python3.9/site-packages/nose/util.py", line 62>
    return func(*arg, **kw)
  File "/build/source/tests/ssl.py", line 414, in test_invalid_country_code
    assert(result.is_error)
AssertionError

----------------------------------------------------------------------
Ran 115 tests in 0.149s

FAILED (failures=1)
```

Comment (RaitoBezarius) [reply pkgs/development/python-modules/ripe-atlas-sagan/default.nix:29]: Indeed, `nixpkgs-review pr <xxx>` triggered the issue.

Comment (RaitoBezarius) [reply pkgs/development/python-modules/ripe-atlas-sagan/default.nix:29]: From what I gather from the logs:
```
tests.test_invalid_country_code ... /build/source/ripe/atlas/sagan/ssl.py:131: UserWarning: Country names should be two characters, but the attribute is 6 characters in length.
  for ext in cert.extensions:
/build/source/ripe/atlas/sagan/ssl.py:79: UserWarning: Country names should be two characters, but the attribute is 6 characters in length.
  if cert and cert.subject:
/build/source/ripe/atlas/sagan/ssl.py:81: UserWarning: Country names should be two characters, but the attribute is 6 characters in length.
  self._parse_x509_name(cert.subject)
/build/source/ripe/atlas/sagan/ssl.py:83: UserWarning: Country names should be two characters, but the attribute is 6 characters in length.
  if cert and cert.issuer:
/build/source/ripe/atlas/sagan/ssl.py:85: UserWarning: Country names should be two characters, but the attribute is 6 characters in length.
  self._parse_x509_name(cert.issuer)
FAIL
```

Comment (RaitoBezarius) [reply pkgs/development/python-modules/ripe-atlas-sagan/default.nix:29]: I disabled the test and marked it as flaky.

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 155267` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python310Packages.ripe-atlas-sagan</li>
    <li>python39Packages.ripe-atlas-sagan</li>
  </ul>
</details>

**Review by fabaff — APPROVED** [review]
Looks good to me.

**Comment by RaitoBezarius** [comment]
Thanks @fabaff for the fixups, I appreciate it!

**Pull request merged**